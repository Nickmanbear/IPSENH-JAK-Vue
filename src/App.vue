<template>
  <div id="app">
    <Navigation />
    <router-view/>
    <img v-bind:src="imgUrl" alt="background image">
  </div>
</template>

<script>
import Navigation from '@/components/Navigation.vue';

export default {
  name: 'App',
  components: {
    Navigation,
  },
  data() {
    return {
      publicPath: process.env.BASE_URL,
      img: '',
    };
  },
  computed: {
    imgUrl() {
      return `${this.publicPath}images/${this.img}`;
    },
  },
  mounted() {
    const images = ['MtFujiWithTemple.jpg', 'MtFujiWithLake.jpg'];

    const count = Math.trunc(Math.random() * 2);
    this.img = images[count];
  },
};
</script>

<style lang="scss">
body {
  margin: 0;
}

#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  /*text-align: center;*/
  color: #2c3e50;

  img {
    z-index: -1;
    position: fixed;
    overflow: hidden;
    /*top: -85%;*/
    top: 0;
    left: 0;
    min-height: 100vh;
    min-width: 100vw;
    opacity: 75%;
    /*background-size: cover;*/
  }
}
</style>
