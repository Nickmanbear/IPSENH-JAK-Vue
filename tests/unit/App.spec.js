import { shallowMount } from '@vue/test-utils';
import App from '@/App.vue';

describe('App', () => {
  let cmp;

  beforeEach(() => {
    cmp = shallowMount(App, {
      stubs: ['router-view'],
      data() {
        return {
          img: '',
        };
      },
    });
    cmp.setData({ img: 'test url' });
  });

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot();
  });
});
