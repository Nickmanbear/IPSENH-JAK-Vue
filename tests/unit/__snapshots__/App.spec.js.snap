// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has the expected html structure 1`] = `
<div
  id="app"
>
  <div
    id="nav"
  >
    <!---->
  </div>
   
  <router-view-stub />
</div>
`;
